app/src/main/java/com/kickstarter/libs/utils/AnalyticEventsUtils.kt

```diff
@@ -0,0 +1,226 @@
+                if (category.isRoot) {
+                    put("category", category.name())
+                } else {
+                    category.parent()?.let { parent ->
+                        put("category", parent.name())
+                    } ?: category.parentName()?.let {
+                        put("category", it)
+                    }
+                    put("subcategory", category.name())
+                }
+            }
+            project.commentsCount()?.let { put("comments_count", it) }
+            put("country", project.country())
+            put("creator_uid", project.creator().id())
+            put("currency", project.currency())
+            put("current_pledge_amount", project.pledged())
+            put("current_amount_pledged_usd", project.pledged() * project.staticUsdRate())
+            project.deadline()?.let { deadline ->
+                put("deadline", deadline.millis / 1000)
+            }
+            put("duration", ProjectUtils.timeInSecondsOfDuration(project).toFloat().roundToInt())
+            put("goal", project.goal())
+            put("goal_usd", project.goal() * project.staticUsdRate())
+            put("has_video", project.video() != null)
+            put("hours_remaining", ceil((ProjectUtils.timeInSecondsUntilDeadline(project) / 60.0f / 60.0f).toDouble()).toInt())
+            put("is_repeat_creator", IntegerUtils.intValueOrZero(project.creator().createdProjectsCount()) >= 2)
+            project.launchedAt()?.let { launchedAt ->
+                put("launched_at", launchedAt.millis / 1000)
+            }
+            project.location()?.let { location ->
+                put("location", location.name())
+            }
+            put("name", project.name())
+            put("percent_raised", project.percentageFunded() / 100.0f)
+            put("pid", project.id())
+            put("prelaunch_activated", BooleanUtils.isTrue(project.prelaunchActivated()))
+            project.rewards()?.let { rewards ->
+                put("rewards_count", rewards.size)
+            }
+            put("state", project.state())
+            put("static_usd_rate", project.staticUsdRate())
+            project.updatesCount()?.let { put("updates_count", it) }
+            put("user_is_project_creator", ProjectUtils.userIsCreator(project, loggedInUser))
+            put("user_is_backer", project.isBacking)
+            put("user_has_watched", project.isStarred)
+        }
+
+        return MapUtils.prefixKeys(properties, prefix)
+    }
+
+    fun refTagProperties(intentRefTag: RefTag?, cookieRefTag: RefTag?) = HashMap<String, Any>().apply {
+                intentRefTag?.tag()?.let { put("session_ref_tag", it) }
+                cookieRefTag?.tag()?.let { put("session_referrer_credit", it) }
+            }
+
+    @JvmOverloads
+    fun activityProperties(activity: Activity, loggedInUser: User?, prefix: String = "activity_"): Map<String, Any> {
+        val props = HashMap<String, Any>()
```

Comment:
you can put initialize this in one line `val props: HashMap<String, Any> = hashMapOf("category" to activity.category())`